python/tests/datasources/test_acs_health_insurance.py

```diff
@@ -0,0 +1,174 @@
+    df = acsHealthInsurance.post_process(df, 'sex', 'state')
+
+    expected_df = pd.read_csv(GOLDEN_DATA_STATE_SEX, dtype={'state_fips': str})
+    cols = list(expected_df.columns)
+    assert_frame_equal(df.sort_values(cols).reset_index(drop=True),
+                       expected_df.sort_values(cols).reset_index(drop=True),
+                       check_like=True)
+
+
+@mock.patch('ingestion.gcs_to_bq_util.load_public_dataset_from_bigquery_as_df',
+            side_effect=get_fips_and_county_names_as_df)
+@mock.patch('ingestion.gcs_to_bq_util.load_values_as_df',
+            side_effect=_get_by_race_as_df)
+def testSexCounty(mock_acs: mock.MagicMock, mock_fips: mock.MagicMock):
+    acsHealthInsurance = AcsHealthInsurance()
+
+    df = acsHealthInsurance.get_raw_data('sex', 'county', get_acs_metadata_as_json(), 'some-bucket')
+    df = acsHealthInsurance.post_process(df, 'sex', 'county')
+
+    expected_df = pd.read_csv(GOLDEN_DATA_COUNTY_SEX, dtype={'state_fips': str, 'county_fips': str})
+    cols = list(expected_df.columns)
+    assert_frame_equal(df.sort_values(cols).reset_index(drop=True),
+                       expected_df.sort_values(cols).reset_index(drop=True),
+                       check_like=True)
+
+
+@mock.patch('ingestion.gcs_to_bq_util.load_public_dataset_from_bigquery_as_df',
+            side_effect=get_fips_and_county_names_as_df)
+@mock.patch('ingestion.gcs_to_bq_util.load_values_as_df',
+            side_effect=_get_by_race_as_df)
+def testRaceCounty(mock_acs: mock.MagicMock, mock_fips: mock.MagicMock):
+    acsHealthInsurance = AcsHealthInsurance()
+
+    df = acsHealthInsurance.get_raw_data('race', 'county', get_acs_metadata_as_json(), 'some-bucket')
+    df = acsHealthInsurance.post_process(df, 'race', 'county')
+
+    expected_df = pd.read_csv(GOLDEN_DATA_COUNTY_RACE, dtype={'state_fips': str, 'county_fips': str})
+    cols = list(expected_df.columns)
+    assert_frame_equal(df.sort_values(cols).reset_index(drop=True),
+                       expected_df.sort_values(cols).reset_index(drop=True),
+                       check_like=True)
+
+
+@mock.patch('ingestion.census.fetch_acs_metadata',
+            return_value=get_acs_metadata_as_json())
+@mock.patch('ingestion.gcs_to_bq_util.load_values_as_df',
+            side_effect=_get_by_race_as_df)
+@mock.patch('ingestion.gcs_to_bq_util.load_public_dataset_from_bigquery_as_df',
+            side_effect=get_fips_and_county_names_as_df)
+@mock.patch('ingestion.gcs_to_bq_util.add_df_to_bq',
+            return_value=None)
+def testWriteToBq(mock_bq: mock.MagicMock,
+                  mock_fips: mock.MagicMock,
+                  mock_acs: mock.MagicMock,
+                  mock_json: mock.MagicMock):
+    acsHealthInsurance = AcsHealthInsurance()
+
+    acsHealthInsurance.write_to_bq('dataset', 'gcs_bucket')
+
+    assert mock_json.call_count == 1
+
+    # One call per race per geo, and then one call for sex at each geo
```

Comment:
thanks for adding these comments in (and the explicit addition) i think it's really makes these assertions more useful because if/when i break it later changing code it's easier for me to understand why lol

Reply:
Thats what tests are for!